packages/web-app-files/src/composables/sort/useSort.ts

```diff
@@ -63,8 +65,13 @@ export function useSort<T extends SortableItem>(options: SortOptions<T>): SortRe
   })
 
   const handleSort = ({ sortBy, sortDir }: { sortBy: string; sortDir: SortDir }) => {
-    sortByRef.value = sortBy
-    sortDirRef.value = sortDir
+    return router.replace({
+      query: {
+        ...unref(router.currentRoute).query,
+        'sort-by': sortBy,
+        'sort-dir': sortDir
+      }
+    })
```

Comment:
This fixes https://github.com/owncloud/web/issues/8409.

We (currently) need to update those params at the same time, otherwise one of the 2 changes will get swallowed by the other. This happens because of the new async behaviour of `vue-router`.

Another, more global approach to fix this can be found here: https://github.com/owncloud/web/pull/8414. But it has other drawbacks, hence I figured we solve it this way - at least for now.

Reply:
Maybe worth putting this comment into source?